WooCommerce/Classes/Tools/BoldableTextParser.swift

```diff
@@ -0,0 +1,69 @@
+import Foundation
+
+/// Describes a text that could be bolded or not.
+struct BoldableElement: Equatable {
+    let content: String
+    let isBold: Bool
+}
+
+/// Parses boldable text for further text concatenation with bold attributes.
+struct BoldableTextParser {
+    private let boldTextPattern = "\\*{2}(.*?)\\*{2}"
+
+    /// Parses the input text and returns a collection of `BoldableElement`.
+    ///
+    /// - Returns: A collection of boldable elements.
+    func parseBoldableElements(string: String) -> [BoldableElement] {
+        guard let regex = try? NSRegularExpression(pattern: boldTextPattern) else {
+            return []
+        }
+        let range = NSRange(string.startIndex..., in: string)
```

Comment:
Nice!
TIL that you can create a range like this. I always used both upper and lower bounds before.